### PR TITLE
Support for specifying the group for a file

### PR DESCRIPTION
**Pull request by sahilTakiar**
Modification to `AvroHdfsDataWriter` that allows a config to specify the group for a file written to HDFS.

### CODE REVIEW

**Comment by sahilTakiar** [comment]
@liyinan926 and @ydailinkedin can you take a look?

**Comment by ydailinkedin** [comment]
LGTM!

File: gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java [hunk 1]
```diff
@@ -219,6 +219,7 @@
   public static final String WRITER_PARTITION_LEVEL = WRITER_PREFIX + ".partition.level";
   public static final String WRITER_PARTITION_PATTERN = WRITER_PREFIX + ".partition.pattern";
   public static final String WRITER_PARTITION_TIMEZONE = WRITER_PREFIX + ".partition.timezone";
+  public static final String WRITER_GROUP_NAME = WRITER_PREFIX + ".owner.name";
```
Comment (liyinan926) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: Should it be named `writer.group.name`?

Comment (sahilTakiar) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: Yes, you are write. Fixed.

**Comment by liyinan926** [comment]
LGTM.

**Comment by dmoore247** [comment]
Interested observer here: The roadmap calls for ORC, Parquet and many other types of HDFS writers I presume. When would it make sense to abstract the file attribute setting code out from the file format code? Put another way, setting the permissions and group ownership is independent of Avro. At some point in the future, setting Hadoop ACLs to facilitate more advanced security needs being required, you would not want to couple that to Avro, vs. ORC vs. Parquet...
Respectfully,
Douglas

**Comment by liyinan926** [comment]
@dmoore247 yes, code that deals with general file attributes should be moved to `FsDataWriter`. We will fix this.

**Pull request merged**